### ISSUE TITLE
Different struct size for measure generic struct in interactive mode vs executable

### ISSUE THREAD

**Issue by zpodlovics** [Microsoft/visualfsharp#668]
Based on the example i posted on #655 I noticed that the interactive mode (fsi) shows different struct sizes for the same source code compared to the executable.

Environment: Ubuntu 14.04 x86_64, F# version: 4.0.0.3-0xamarin1, Mono version: Stable 4.0.4.1/5ab4c0d Tue Aug 25 23:11:51 UTC 2015

Example code:
```fsharp
open System.Runtime.InteropServices

#nowarn "9"

[<Measure>]
type Version

[<Struct>]
[<NoEquality;NoComparison>]
type S =
  val X: int

[<Struct>]
[<NoEquality;NoComparison>]
type SM =
  val X: int<Version> 

[<Struct>]
[<NoEquality;NoComparison>]
type SMG<[<Measure>] 'u> =
  val X: int<'u> 

[<Struct; StructLayout(LayoutKind.Sequential, Pack=1)>]
[<NoEquality;NoComparison>]
type SS =
  val X: int

[<Struct; StructLayout(LayoutKind.Sequential, Pack=1)>]
[<NoEquality;NoComparison>]
type SMS =
  val X: int<Version> 

[<Struct; StructLayout(LayoutKind.Sequential, Pack=1)>]
[<NoEquality;NoComparison>]
type SMGS<[<Measure>] 'u> =
  val X: int<'u> 

let f (x : 'T when 'T: unmanaged and 'T: struct) = printfn "%A" x

f (S())
f (SM())
//f (SMG()) // FS0001: A generic construct requires that the type 'SMG<'u>' is an unmanaged type
//f (SMG<Version>()) // FS0001: A generic construct requires that the type 'SMG<Version>' is an unmanaged type
f (SS())
f (SMS())
//f (SMGS()) // FS0001: A generic construct requires that the type 'SMGS<'u>' is an unmanaged type
//f (SMGS<Version>()) // FS0001: A generic construct requires that the type 'SMGS<Version>' is an unmanaged type

printf "is 64 bit process: %A\n" (System.Environment.Is64BitProcess)
printf "pointer size in bytes: %A\n" (System.IntPtr.Size)

printf "is SMG<Version> a generic type: %A\n" (typeof<SMG<Version>>.IsGenericType)
printf "is SMG<Version> a value type: %A\n" (typeof<SMG<Version>>.IsValueType)

printf "sizeof<S>: %A\n" (sizeof<S>)
printf "sizeof<SM>: %A\n" (sizeof<SM>)
printf "sizeof<SMG<Version>>: %A\n" (sizeof<SMG<Version>>)

printf "sizeof<SS>: %A\n" (sizeof<SS>)
printf "sizeof<SMS>: %A\n" (sizeof<SMS>)
printf "sizeof<SMGS<Version>: %A\n" (sizeof<SMGS<Version>>)

printf "Marshal.SizeOf(typeof<S>): %A\n" (Marshal.SizeOf(typeof<S>))
printf "Marshal.SizeOf(typeof<SM>): %A\n" (Marshal.SizeOf(typeof<SM>))
printf "Marshal.SizeOf(sizeof<SMG<Version>>): %A\n" (Marshal.SizeOf(typeof<SMG<Version>>))

printf "Marshal.SizeOf(typeof<SS>): %A\n" (Marshal.SizeOf(typeof<SS>))
printf "Marshal.SizeOf(typeof<SMS>): %A\n" (Marshal.SizeOf(typeof<SMS>))
printf "Marshal.SizeOf(typeof<SMGS<Version>): %A\n" (Marshal.SizeOf(typeof<SMGS<Version>>))

```

The compiled version (exe) does not show the same result when it executed from fsi:

Compiled version:
```
Structgen+S
Structgen+SM
Structgen+SS
Structgen+SMS
is 64 bit process: true
pointer size in bytes: 8
is SMG<Version> a generic type: false
is SMG<Version> a value type: true
sizeof<S>: 4
sizeof<SM>: 4
sizeof<SMG<Version>>: 4
sizeof<SS>: 4
sizeof<SMS>: 4
sizeof<SMGS<Version>: 4
Marshal.SizeOf(typeof<S>): 4
Marshal.SizeOf(typeof<SM>): 4
Marshal.SizeOf(sizeof<SMG<Version>>): 4
Marshal.SizeOf(typeof<SS>): 4
Marshal.SizeOf(typeof<SMS>): 4
Marshal.SizeOf(typeof<SMGS<Version>): 4
```

fsi session (fsharpi --load:structgen.fsx)

```
FSI_0002.Structgen+S
FSI_0002.Structgen+SM
FSI_0002.Structgen+SS
FSI_0002.Structgen+SMS
is 64 bit process: true
pointer size in bytes: 8
is SMG<Version> a generic type: false
is SMG<Version> a value type: true
sizeof<S>: 4
sizeof<SM>: 4
sizeof<SMG<Version>>: 8
sizeof<SS>: 4
sizeof<SMS>: 4
sizeof<SMGS<Version>: 8
Marshal.SizeOf(typeof<S>): 4
Marshal.SizeOf(typeof<SM>): 4
Marshal.SizeOf(sizeof<SMG<Version>>): 8
Marshal.SizeOf(typeof<SS>): 4
Marshal.SizeOf(typeof<SMS>): 4
Marshal.SizeOf(typeof<SMGS<Version>): 8
```

IL representation:
type S
```
  .class nested public sequential ansi sealed serializable S
  	extends [mscorlib]System.ValueType
  {
    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.StructAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoEqualityAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoComparisonAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) =  (01 00 03 00 00 00 00 00 ) // ........

    .field  assembly  int32 X@

    // method line 4
    .method public hidebysig specialname 
           instance default int32 get_X ()  cil managed 
    {
        // Method begins at RVA 0x208c
	// Code size 7 (0x7)
	.maxstack 8
	IL_0000:  ldarg.0 
	IL_0001:  ldfld int32 Structgen/S::X@
	IL_0006:  ret 
    } // end of method S::get_X

	.property instance int32 X ()
	{
		.custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags, int32) =  (01 00 04 00 00 00 00 00 00 00 00 00 ) // ............

		.get instance default int32 Structgen/S::get_X () 
	}
  } // end of class S
```

type SM
```
  .class nested public sequential ansi sealed serializable SM
  	extends [mscorlib]System.ValueType
  {
    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.StructAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoEqualityAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoComparisonAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) =  (01 00 03 00 00 00 00 00 ) // ........

    .field  assembly  int32 X@

    // method line 5
    .method public hidebysig specialname 
           instance default int32 get_X ()  cil managed 
    {
        // Method begins at RVA 0x2094
	// Code size 7 (0x7)
	.maxstack 8
	IL_0000:  ldarg.0 
	IL_0001:  ldfld int32 Structgen/SM::X@
	IL_0006:  ret 
    } // end of method SM::get_X

	.property instance int32 X ()
	{
		.custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags, int32) =  (01 00 04 00 00 00 00 00 00 00 00 00 ) // ............

		.get instance default int32 Structgen/SM::get_X () 
	}
  } // end of class SM
```

type SMG
```
  .class nested public sequential ansi sealed serializable SMG
  	extends [mscorlib]System.ValueType
  {
    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.StructAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoEqualityAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.NoComparisonAttribute::'.ctor'() =  (01 00 00 00 ) // ....

    .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) =  (01 00 03 00 00 00 00 00 ) // ........

    .field  assembly  int32 X@

    // method line 6
    .method public hidebysig specialname 
           instance default int32 get_X ()  cil managed 
    {
        // Method begins at RVA 0x209c
	// Code size 7 (0x7)
	.maxstack 8
	IL_0000:  ldarg.0 
	IL_0001:  ldfld int32 Structgen/SMG::X@
	IL_0006:  ret 
    } // end of method SMG::get_X

	.property instance int32 X ()
	{
		.custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags, int32) =  (01 00 04 00 00 00 00 00 00 00 00 00 ) // ............

		.get instance default int32 Structgen/SMG::get_X () 
	}
  } // end of class SMG
```

unified diff between S and SMG
```diff
--- S.il	2015-10-05 17:45:21.402557379 +0200
+++ SMG.il	2015-10-05 17:45:40.866749017 +0200
@@ -1,4 +1,4 @@
-  .class nested public sequential ansi sealed serializable S
+  .class nested public sequential ansi sealed serializable SMG
   	extends [mscorlib]System.ValueType
   {
     .custom instance void class [FSharp.Core]Microsoft.FSharp.Core.StructAttribute::'.ctor'() =  (01 00 00 00 ) // ....
@@ -11,22 +11,22 @@
 
     .field  assembly  int32 X@
 
-    // method line 4
+    // method line 6
     .method public hidebysig specialname 
            instance default int32 get_X ()  cil managed 
     {
-        // Method begins at RVA 0x208c
+        // Method begins at RVA 0x209c
 	// Code size 7 (0x7)
 	.maxstack 8
 	IL_0000:  ldarg.0 
-	IL_0001:  ldfld int32 Structgen/S::X@
+	IL_0001:  ldfld int32 Structgen/SMG::X@
 	IL_0006:  ret 
-    } // end of method S::get_X
+    } // end of method SMG::get_X
 
 	.property instance int32 X ()
 	{
 		.custom instance void class [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::'.ctor'(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags, int32) =  (01 00 04 00 00 00 00 00 00 00 00 00 ) // ............
 
-		.get instance default int32 Structgen/S::get_X () 
+		.get instance default int32 Structgen/SMG::get_X () 
 	}
-  } // end of class S
+  } // end of class SMG

```

Maybe a value type compilation / representation bug in interactive mode?

**Comment by zpodlovics** [comment]
It seems that this issue is fixed by #656

**Comment by dsyme** [comment]
Yes, fixed by https://github.com/Microsoft/visualfsharp/pulls/656
